### PR TITLE
[CARBONDATA-2338][Test] Add example to upload data to S3 by using SDK

### PR DESCRIPTION
**Pull request by xubo245**
Add example to show:  
      carbondata can write SDK's unmanage/transactional table data to S3

And then we can read data from S3 by external table.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4968/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4970/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4420/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4421/

**Comment by xubo245** [comment]
@ajantha-bhat @jackylk Please review it.

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+package org.apache.carbondata.examples
+
+import org.apache.hadoop.fs.s3a.Constants.{ACCESS_KEY, ENDPOINT, SECRET_KEY}
+import org.apache.spark.sql.SparkSession
+import org.slf4j.{Logger, LoggerFactory}
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.sdk.file.{CarbonWriter, Schema}
+
+object S3UsingSDKExample {
```
Comment (jackylk) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala]: please add comment to describe this example

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  // prepare SDK writer output
+  def buildTestData(
+      path: String,
+      num: Int = 3,
+      persistSchema: Boolean = false): Any = {
+
+    // getCanonicalPath gives path with \, so code expects /. Need to handle in code ?
+    val writerPath = path.replace("\\", "/");
+
+    val schema = new StringBuilder()
```
Comment (jackylk) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala]: To make it simpler, can you use the object way to construct the schema object

Comment (xubo245) [reply examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala]: I refer merged code in carbondata, there are many usage like that: 

org.apache.carbondata.spark.testsuite.createTable.TestCarbonFileInputFormatWithExternalCarbonTable

org.apache.carbondata.spark.testsuite.createTable.TestUnmanagedCarbonTable

Keep it? or change all similar place in carbondata?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3878/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5100/

**Comment by xubo245** [comment]
retest this please

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5118/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3897/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3901/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5122/

**Comment by xubo245** [comment]
@jackylk  CI pass

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3943/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5228/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala (hunk 2 above)
Comment (jackylk) [reply examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala]: please refer to CSVCarbonWriterTest.testWriteFiles

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4079/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5260/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala (hunk 2 above)
Comment (xubo245) [reply examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala]: ok,done

**Comment by jackylk** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4149/

**Pull request closed without merging**